File: Find-Minimum-in-Rotated-Sorted-Array/Find-Minimum-in-Rotated-Sorted-Array.py
nums = [3,4,5,1,2]

def findMin(nums):
    l =0
    r = len(nums)-1
    while l<r:
        m = l+(r-l)//2
        if nums[r]<nums[m]:
            l = m +1
        else:
            r =m
    return nums[r]

print(findMin(nums))
    

File: Find-Minimum-in-Rotated-Sorted-Array/Find-Minimum-in-Rotated-Sorted-Array.js
let nums = [3,1,2]

var findMin = function(nums) {
    let l = 0
    let r = nums.length - 1
    while(l<r){
        let m = Math.floor(r - (r-l)/2)
        if (nums[r]<nums[m]){
            l = m+1
        }else{
            r=m
        }
    }
    return nums[r]
    // nums = nums.sort((a,b)=>a-b)
    // return nums[0]
};

console.log(findMin(nums))